PVRTexLibNET/PVRTexLibNET.cs

```diff
@@ -92,49 +92,102 @@ public enum CompressorQuality
+        public static extern uint GetTextureDataSize(IntPtr pPvrTexture, int iMIPLevel = -1);
 
         [DllImport(dllName, CallingConvention = CallingConvention.Cdecl)]
-        public static extern void GetTextureData(IntPtr data, uint dataSize, uint uiMIPLevel = 0);
+        public static extern void GetTextureData(IntPtr pPvrTexture, IntPtr data, uint dataSize, uint uiMIPLevel = 0);
+        #endregion Interop
+
+        private IntPtr _pPvrTexture = IntPtr.Zero;
 
-        public static void GetTextureData<T>(T[] data, uint dataSize, uint uiMIPLevel = 0)
+        public IntPtr PvrTexturePointer { get { return _pPvrTexture; } }
+        
+        public PVRTexture(string filePath)
+        {
+            IntPtr pPvrTexture = CreateTexture(filePath);
+        }
+        
+        public static PVRTexture CreateTexture<T>(T[] data, uint u32Width, uint u32Height, uint u32Depth, PixelFormat ptFormat, bool preMultiplied, VariableType eChannelType, ColourSpace eColourspace) where T : struct
         {
             var gcHandle = GCHandle.Alloc(data, GCHandleType.Pinned);
-            GetTextureData(gcHandle.AddrOfPinnedObject(), dataSize, uiMIPLevel);
+            var pPvrTexture = CreateTexture(gcHandle.AddrOfPinnedObject(), u32Width, u32Height, u32Depth, ptFormat, preMultiplied, eChannelType, eColourspace);
             gcHandle.Free();
+            return new PVRTexture(pPvrTexture);
+        }
+
+        internal PVRTexture(IntPtr pPvrTexture)
+        {
+            this._pPvrTexture = pPvrTexture;
+        }
+        
+        public bool SaveTexture(string filePath)
+        {
+            return SaveTexture(_pPvrTexture, filePath);
+        }
+
+        public bool Resize(uint u32NewWidth, uint u32NewHeight, uint u32NewDepth, ResizeMode eResizeMode)
+        {
+            return Resize(_pPvrTexture, u32NewWidth, u32NewHeight, u32NewDepth, eResizeMode);
+        }
+        
+        public bool GenerateMIPMaps(ResizeMode eFilterMode, uint uiMIPMapsToDo = int.MaxValue)
+        {
+            return GenerateMIPMaps(_pPvrTexture, eFilterMode, uiMIPMapsToDo);
+        }
+
+        public bool Transcode(PixelFormat ptFormat, VariableType eChannelType, ColourSpace eColourspace, CompressorQuality eQuality = CompressorQuality.PVRTCNormal, bool bDoDither = false)
+        {
+            return Transcode(_pPvrTexture, ptFormat, eChannelType, eColourspace, eQuality, bDoDither);
+        }
+
+        public uint GetTextureDataSize(int iMIPLevel = -1)
+        {
+            return GetTextureDataSize(_pPvrTexture, iMIPLevel);
+        }
+
+        public void GetTextureData<T>(T[] data, uint dataSize, uint uiMIPLevel = 0) where T : struct
+        {
+            var gcHandle = GCHandle.Alloc(data, GCHandleType.Pinned);
+            GetTextureData(_pPvrTexture, gcHandle.AddrOfPinnedObject(), dataSize, uiMIPLevel);
+            gcHandle.Free();
+        }
+
+        #region Implement IDisposable
+        public void Dispose()
```

Comment:
The proper Disposable pattern should be implemented.
- Provide a void Dispose(bool disposing) method that does the release of unmanaged resources, the same as your Dispose() does here.
- Dispose() becomes
````
    public void Dispose()
    {
        Dispose(true);
        GC.SuppressFinalize(this);
    }
````
- Provide a finalizer that calls Dispose(false).